packages/zipkin-instrumentation-superagent/src/superagentPlugin.js

```diff
@@ -19,26 +19,38 @@ const {Instrumentation} = require('zipkin');
 const plugin = ({tracer, serviceName, remoteServiceName}) => {
   const instrumentation = new Instrumentation.HttpClient({tracer, serviceName, remoteServiceName});
 
-  return (req) => tracer.scoped(() => {
+  return (req) => {
     const {method, url} = req;
-    const {headers} = instrumentation.recordRequest(req, url, method);
-    // must get id ONLY AFTER recordRequest
-    const traceId = tracer.id;
 
-    const recordResponse = (res) => {
+    // capture the trace ID from recording the start of the request
+    let traceId;
+    const {headers} = tracer.scoped(() => {
+      const injected = instrumentation.recordRequest(req, url, method);
+      traceId = tracer.id;
+      return injected;
+    });
+
+    // We can't use end() as the caller might be using it.
```

Comment:
we should probably do another cleanup pass later, if we can determine how to get the response and error callbacks at the same time without using end()